src/components/List/List.jsx

```diff
@@ -1,22 +1,59 @@
-import React, { useState } from 'react';
+import React, { useEffect, useState } from 'react';
 import { useNavigate } from 'react-router-dom';
-import db from '../lib/firebase';
-import { collection, doc, updateDoc } from 'firebase/firestore';
+import db from '../../lib/firebase';
+import { collection, doc, updateDoc, query, orderBy } from 'firebase/firestore';
 import { useCollection } from 'react-firebase-hooks/firestore';
 import moment from 'moment';
-
-import { getEstimate, calcTimeDiff, formatDate } from '../helpers';
+import './list.css';
+import {
+  getEstimate,
+  calcTimeDiff,
+  formatDate,
+  calcDaysSince,
+} from '../../helpers';
 
 export const List = ({ token }) => {
   let navigate = useNavigate();
-  const [value, loading, error] = useCollection(collection(db, token), {
+  const q = query(collection(db, token), orderBy('item'));
+  const [value, loading, error] = useCollection(q, {
     snapshotListenOptions: { includeMetadataChanges: true },
   });
+
   let [filterText, setFilterText] = useState('');
+  let [items, setItems] = useState([]);
+
+  useEffect(() => {
+    if (value) {
+      let arr = [];
+      value.docs.forEach((item) => {
+        arr.push({ ...item.data(), id: item.id });
+      });
+      arr.map((item) => (item.isActive = handleActive(item)));
+      arr.sort(
+        (itemA, itemB) =>
+          itemA.estimated_next_purchase - itemB.estimated_next_purchase,
+      );
+      arr.sort(
+        (itemA, itemB) => Number(itemB.isActive) - Number(itemA.isActive),
+      );
+      setItems(arr);
+    }
+  }, [value]);
+
+  const handleActive = (item) => {
+    if (
+      calcDaysSince(item.last_purchased_date) >
+      item.estimated_next_purchase * 2
+    ) {
+      return false;
+    }
+    return true;
+  };
```

Comment:
Beautiful equation. Translating the requirements into this logic, I'm sure wasn't the easiest. Great work! 🎩

Reply:
I agree, this logic is perfect. 👌 Because you're just returning a boolean result, you could even simplify it a bit:

```suggestion

  const handleActive = (item) => {
    return calcDaysSince(item.last_purchased_date) <=
      item.estimated_next_purchase * 2;
  };
```

Notice that I flipped the operator from `>` to `<=` because in the original, when the condition was true you returned false and vice versa, and now I'm just returning the actual value of the condition.